test/functional/cursor.test.js

```diff
@@ -3901,6 +3901,42 @@ describe('Cursor', function () {
     });
   });
 
+  // NODE-2035
+  it('should propagate error when exceptions are thrown from an awaited forEach call', function () {
+    const configuration = this.configuration;
+    const client = configuration.newClient({ w: 1 }, { maxPoolSize: 1 });
+    const docs = [{ a: 1 }, { a: 2 }, { a: 3 }];
+    let collection;
+    let db;
+    return client
+      .connect()
+      .then(() => {
+        this.defer(() => client.close());
+        db = client.db(configuration.db);
+        collection = db.collection('cursor_session_tests2');
+        return collection.insertMany(docs);
+      }, console.error)
+      .then(() => {
+        const cursor = collection.find({});
+        this.defer(() => cursor.close());
+        expect(() => {
+          cursor
+            .forEach(() => {
+              throw new Error('FAILURE IN FOREACH CALL');
+            })
+            .then(
+              () => {
+                expect(false).to.equal(true, 'Failed to catch error thrown in awaited forEach');
+              },
+              err => {
+                expect(err.message).to.deep.equal('FAILURE IN FOREACH CALL');
+              }
+            );
+        }).to.not.throw();
+      })
+      .catch(console.error);
```

Comment:
Are the console usages here for debugging? We should merge assertions if we want to catch something here.

Reply:
I had forgotten to remove the console.error since I had put in there when running into so issues when writing the test so I had it there just so I could figure out where errors were coming from. As far as I know, it is no longer needed.